src/python/pants/build_graph/address.py

```diff
@@ -627,14 +630,25 @@ class BuildFileAddress:
 
 
 class ResolveError(MappingError):
-    """Indicates an error resolving target addresses."""
+    """Indicates an error resolving targets."""
 
     @classmethod
     def did_you_mean(
-        cls, *, bad_name: str, known_names: Iterable[str], namespace: str
+        cls,
+        bad_address: Address,
+        *,
+        description_of_origin: str,
+        known_names: Iterable[str],
+        namespace: str,
     ) -> ResolveError:
-        possibilities = "\n  ".join(f":{target_name}" for target_name in sorted(known_names))
         return cls(
-            f"'{bad_name}' was not found in namespace '{namespace}'. Did you mean one "
-            f"of:\n  {possibilities}"
+            softwrap(
+                f"""
+                The address {bad_address} from {description_of_origin} does not exist.
+
+                The target name ':{bad_address.target_name}' is not defined in the directory
+                {namespace}. Did you mean one of these target names?\n
+                """
+                + bullet_list(f":{name}" for name in known_names)
+            )
```

Comment:
I prefer the other way to formulate this. That is, simply state the known target names and leave the question of the message.